src/core.c/IO/Path.pm6

```diff
@@ -682,27 +684,38 @@ my class IO::Path is Cool does IO {
         # check if we need a BOM
         if $enc eq 'utf16' {
 
-           # add a BOM if writing starts at beginning of file
-           if $mode eq 'w'
-             || nqp::not_i(nqp::stat($path,nqp::const::STAT_EXISTS))
-             || nqp::not_i(nqp::stat($path,nqp::const::STAT_FILESIZE)) {
+            # add a BOM if (over)writing 
+            if $mode eq 'w' {
                 nqp::unshift_i($blob,254);
                 nqp::unshift_i($blob,255);
-           }
+            }
+            # or appending to a new or existing, but zero-length, file
+            else {
+                my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s($path), 0);
+                if nqp::not_i(nqp::dispatch("boot-syscall", "stat-exists", $stat)) ||
+                   nqp::not_i(nqp::dispatch("boot-syscall", "stat-filesize", $stat)) {
+                    nqp::unshift_i($blob,254);
+                    nqp::unshift_i($blob,255);
+                }
+            }
         }
 
         spurt-blob($path, $mode, $blob)
     }
 
     method user(IO::Path:D:) {
-        Rakudo::Internals.FILETEST-E(my str $path = self.absolute)
-          ?? nqp::stat($path,nqp::const::STAT_UID)
+        my str $path = self.absolute;
```

Comment:
Probably doesn't matter, but I wonder if it would be slightly better to call `self.absolute` in void context (instead of assigning to `$path`) to populate `$!os-path`, and then to just us `$!os-path` instead of `$path`

Reply:
Better how/why?

Reply:
No longer need to create $path or assign to it

Reply:
FWIW, given this test script: `MVM_SPESH_BLOCKING=1 raku -e 'my $a = "VERSION".IO; my $b; my $s := now; $b := $a.user for ^1_000_000; say now - $s; say $b'`
* This PR as is reports ~1.25s.
* Changing `my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s($path), 0);` to `my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s(self.absolute), 0);` gives ~1.13s.
* Changing `my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s($path), 0);` instead to `self.absolute; my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s($!os-path), 0);` give ~1.14.

I originally didn't make that change (just using `self.absolute` instead of assigning it to a `$path`) to minimize the diff, but seeing the performance improvement, I'll add a new commit doing it everywhere.